src/converters/toZigbee.ts

```diff
@@ -560,6 +561,27 @@ export const warning: Tz.Converter = {
             {startwarninginfo: info, warningduration: values.duration, strobedutycycle: values.strobeDutyCycle, strobelevel: values.strobeLevel},
             utils.getOptions(meta.mapped, entity),
         );
+
+        // Track and maintain a virtual state of the siren
+        // is on if duration set, and either strobe with dutycycle or alarm is set
+        const virtualState = meta.mapped.meta?.virtualState?.warning;
```

Comment:
What holds us from not enabling this for every device?

Reply:
I simply have no clue how other warning devices return their state ... as this siren doesn't support most of the options, it might not be the best example device to generalize

Reply:
I think it conforms against the ZCL spec so should be fine, let's try it.

Reply:
Made available to all warning devices🤞
The ``siren_state`` will still needs to be exposed